docs/man/rpm-dependency-generators.7.scd

```diff
@@ -0,0 +1,381 @@
+RPM-DEPENDENCY-GENERATORS(7)
+
+# NAME
+*rpm-dependency-generators* - RPM file attribute and dependency generator
+interface
+
+# SYNOPSIS
+
+## Generators
+*%\_\_*_NAME_*\_conflicts*[*()*] _BODY_++
+*%\_\_*_NAME_*\_enhances*[*()*] _BODY_++
+*%\_\_*_NAME_*\_obsoletes*[*()*] _BODY_++
+*%\_\_*_NAME_*\_orderwithrequires*[*()*] _BODY_++
+*%\_\_*_NAME_*\_provides*[*()*] _BODY_++
+*%\_\_*_NAME_*\_recommends*[*()*] _BODY_++
+*%\_\_*_NAME_*\_requires*[*()*] _BODY_++
+*%\_\_*_NAME_*\_suggests*[*()*] _BODY_++
+*%\_\_*_NAME_*\_supplements*[*()*] _BODY_
+
+## Matching rules
+*%\_\_*_NAME_*\_path* _REGEX_++
+*%\_\_*_NAME_*\_magic* _REGEX_++
+*%\_\_*_NAME_*\_exclude_path* _REGEX_++
+*%\_\_*_NAME_*\_exclude_magic* _REGEX_
+
+## Tunables
+*%\_\_*_NAME_*\_conflicts_opts* _BODY_++
+*%\_\_*_NAME_*\_enhances_opts* _BODY_++
+*%\_\_*_NAME_*\_obsoletes_opts* _BODY_++
+*%\_\_*_NAME_*\_orderwithrequires_opts* _BODY_++
+*%\_\_*_NAME_*\_provides_opts* _BODY_++
+*%\_\_*_NAME_*\_recommends_opts* _BODY_++
+*%\_\_*_NAME_*\_requires_opts* _BODY_++
+*%\_\_*_NAME_*\_suggests_opts* _BODY_++
+*%\_\_*_NAME_*\_supplements_opts* _BODY_
+
+*%\_\_requires_exclude* _REGEX_++
+*%\_\_requires_exclude_from* _REGEX_++
+*%\_\_provides_exclude* _REGEX_++
+*%\_\_provides_exclude_from* _REGEX_
+
+# DESCRIPTION
+
+RPM uses a pluggable interface coupled with a file classifier to add dependency
+information to packages at build time. This system relies on specialized, domain
+specific executables, called _generators_, that do the heavy-lifting of figuring
+out the dependencies from the files present in the buildroot. It complements the
+manual tags, such as *Requires* or *Provides*, in *rpm-spec*(5) files.
+
+Generators are paired with logical labels, called _file attributes_, assigned to
+the packaged files by the file classifier. This ensures that every file is only
+processed by the applicable generator(s), as opposed to all generators, which
+simplifies their implementation and significantly improves the overall build
+performance.
+
+# FILE ATTRIBUTES
+
+A file attribute is a logical label that RPM gives to one or more files in the
+buildroot, based on a set of matching _rules_. It can represent a single file
+type, such as ELF, or a broader domain, such as all source files written in a
+specific programming language.
+
+These attributes are distinct from the file attributes in the filesystem and are
+only kept in memory during package builds. A single file may have an arbitrary
+number of attributes, and a single attribute may be assigned to an arbitrary
+number of files.
+
+A valid file attribute called _NAME_ is introduced with an *rpm-macrofile*(5),
+_NAME.attr_, dropped into the *%{\_fileattrsdir}* directory, that defines at
+least one of the following macros:
+
+```
+%__NAME_path
+%__NAME_magic
+%__NAME_exclude_path
+%__NAME_exclude_magic
```

Comment:
I think the excludes should be left out from here - it doesn't make much sense to supply a fileattr that only contains excludes.

Reply:
Oh, and just realized: %__NAME_mime should be included in this list.

Reply:
Maybe convert just one for starters and then we can ponder the options?

Reply:
Heh, I was about to say I'm not really sure the ` *%\_\_*_NAME_*\` mess is more understandable and readable than just `*%\_\_NAME_path*` but trying them side-by-side the latter just doesn't seem right, because it does break the style guideline of bold meaning literal.

Reply:
I did end up converting those code blocks yesterday (in my local branch) but the config style actually might be even better, so I'll try it out quickly, I guess. Thanks for the tip :+1:

Reply:
> Heh, I was about to say I'm not really sure the ` *%\_\_*_NAME_*\` mess is more understandable and readable than just `*%\_\_NAME_path*` but trying them side-by-side the latter just doesn't seem right, because it does break the style guideline of bold meaning literal.

Yeah, I'm not a huge fan of the underline style combined with literal underlines, it makes it super awkward to type (and read also, to a degree), although it does follow our own guidelines and is more consistent with the other man pages, so there's that...